include/seqan3/alphabet/cigar/cigar.hpp

```diff
@@ -182,6 +182,35 @@ class cigar : public alphabet_tuple_base<cigar, uint32_t, exposition_only::cigar
 
         return *this;
     }
+
+    /*!\brief Assign from a contigous char array.
+     * \details
+     *
+     * In order to avoid unnecessary copies, you can initialise a seqan3::cigar from two char pointers that point to a
+     * contigous char array that stores the cigar string.
+     *
+     * \include test/snippet/alphabet/cigar/cigar_assign_string.cpp
+     *
+     * \experimentalapi{Experimental since version 3.2.}
+     */
+    cigar & assign_string(char const * start, char const * end) noexcept
+    {
+        uint32_t num{};
+        auto [ ptr, errc ] = std::from_chars(start, end, num);
+
+        if ((errc != std::errc{}) || (!char_is_valid_for<operation>(*ptr)))
```

Comment:
Is it deliberate that whenever `std::from_chars` fails, we just return `0P`?

samtools just throws 
```
[E::parse_cigar] Unrecognized CIGAR operator
[W::sam_read1_sam] Parse error at line 65
samtools view: error reading file "test.sam"
```

And the specs specify a specific set of possible operations.
We probably had a reason, but the reason should go into the documentation.

Reply:
see documentation

Reply:
I still think this is strange. While "0P" might not lead to an error, assign "WhatTheF" to a Cigar **should** probably cause an error, or not?

Reply:
I also think this should, give "0F" (sorry for interrupting couldn't pass this opportunity.)

Reply:
I guess it's ok for now (outside of the scope of this pr). But we might want to revisit that...